.github/workflows/build-push.yml

```diff
@@ -0,0 +1,366 @@
+name: Build and push assets
+
+on:
+  workflow_call:
+    inputs:
+      NODE_OPTIONS:
+        description: Space-separated list of command-line Node options.
+        type: string
+        default: ''
+        required: false
+      NODE_VERSION:
+        description: Node version to be used.
+        default: 18
+        required: false
+        type: string
+      NPM_REGISTRY_DOMAIN:
+        description: Domain of the private npm registry.
+        default: https://npm.pkg.github.com/
+        required: false
+        type: string
+      PHP_VERSION:
+        description: PHP version with which the PHP tools are to be executed.
+        default: '8.2'
+        required: false
+        type: string
+      PHP_TOOLS:
+        description: PHP tools supported by shivammathur/setup-php to be installed.
+        default: ''
+        required: false
+        type: string
+      COMPOSER_ARGS:
+        description: Set of arguments passed to Composer.
+        default: '--no-dev --prefer-dist --optimize-autoloader'
+        required: false
+        type: string
+      PACKAGE_NAME:
+        description: The name of the package (falls back to the repository name).
+        required: false
+        default: ''
+        type: string
+      PACKAGE_VERSION:
+        description: The new package version. If not provided, will use latest release version with branch name as pre-release identifier.
+        required: false
+        type: string
+      PRE_SCRIPT:
+        description: Run custom shell code before creating the release archive.
+        default: ''
+        required: false
+        type: string
+      BUILT_BRANCH_NAME:
+        description: Sets the target branch for pushing assets. If not provided, strips 'dev/' prefix from origin branch.
+        type: string
+        default: ''
+        required: false
+
+    secrets:
+      COMPOSER_AUTH_JSON:
+        description: Authentication for privately hosted packages and repositories as a JSON formatted object.
+        required: false
+      NPM_REGISTRY_TOKEN:
+        description: Authentication for the private npm registry.
+        required: false
+      GITHUB_USER_EMAIL:
+        description: Email address for the GitHub user configuration.
+        required: true
+      GITHUB_USER_NAME:
+        description: Username for the GitHub user configuration.
+        required: true
+      GITHUB_USER_SSH_KEY:
+        description: Private SSH key associated with the GitHub user passed as `GITHUB_USER_NAME`.
+        required: true
+      GITHUB_USER_SSH_PUBLIC_KEY:
+        description: Public SSH key associated with the GitHub user passed as `GITHUB_USER_NAME`.
+        required: true
+      ENV_VARS:
+        description: Additional environment variables as a JSON formatted object.
+        required: false
+
+jobs:
+  build-push:
+    timeout-minutes: 10
+    runs-on: ubuntu-latest
+    env: # Used to check whether there are changes to commit
```

Comment:
Just a small note, but can we keep comments consistently always above the line it is affecting? 

```yml
env: 
	# Used to check whether there are changes to commit
	HAS_GIT_CHANGES: ''
	# Disables symlinking of local "path" repositories during package installation.
	# During development, symlinking is preferable.
	# In production builds, we ship the actual install location and remove the source directory.
	COMPOSER_MIRROR_PATH_REPOS: 1
	# Will be overridden by determine package version step
	PACKAGE_VERSION: ${{ inputs.PACKAGE_VERSION }}
	PACKAGE_NAME: ${{ inputs.PACKAGE_NAME && inputs.PACKAGE_NAME || github.event.repository.name }}
```